### PR TITLE
Accessibility enhancement: migrate to use html dialog AB#1491978

### LINKED ISSUE #133: Accessibility bug: focus goes behind drawer when no button is present
### What browser client(s) are you seeing the issue on?

- [x] Firefox
- [x] Chrome
- [x] Safari
- [x] Microsoft Edge

### On which platforms are you seeing the issue?

- [x] Desktop
- [x] Mobile

### Steps to reproduce

1. Create a auro-drawer
2. Include some form field elements such as 2 auro-inputs.
3. Include a disabled auro-button or no auro-button at all. This step is important.
4. Once the dialog is opened, tab thru the dialog field

### Expected behavior

_No response_

### Which version of auro-drawer do you have installed?

latest

### Reproducible on the Auro doc site

- [x] I am able to reproduce this issue on the [Auro doc site](https://auro.alaskaair.com).

### Exit criteria (To be determined by the Auro team)

_No response_

### PR DESCRIPTION
**Pull request by rmenner**
# Alaska Airlines Pull Request

### [AB#1491977](https://itsals.visualstudio.com/5e9f12eb-f830-406f-bee9-be25938f7aaa/_workitems/edit/1491977)

**Links:**
- [AB#1491977](https://itsals.visualstudio.com/E_Retain_Content/_workitems/edit/1491977) — Azure DevOps ticket

This branch will implement the changes outlined in the TRD for this bug fix.
TRD: https://github.com/orgs/AlaskaAirlines/discussions/624

## Review Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

<details>
<summary>
RC Checklist
</summary>

### Browsers

[Browsers Support Guide](https://auro.alaskaair.com/support/browsersSupport)

[Dev demo link](https://alaskaairlines.github.io/auro-drawer/)

Android

- [ ] Chrome
- [ ] Firefox

 iOS

- [ ] Chrome
- [ ] Firefox
- [ ] Safari

Desktop

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### Scenarios

- [ ] Validated linked issues with issue reporting team
- [ ] Test coverage report review

[Framework playground ](https://github.com/AlaskaAirlines/auro-framework-playground)

- [ ] Next React
- [ ] SvelteKit

</details><br>
**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

## Summary by Sourcery

Migrate auro-drawer to native dialog behavior to provide reliable accessibility, focus management, dismissal handling, and scroll control across drawer variants.

New Features:
- Migrate drawer presentation and accessibility behavior to native HTML dialog semantics for modal, non-modal, and nested drawers.
- Add CSS custom property controls for native dialog backdrop styling.
- Add a keyboard and assistive-technology audit demo covering focus, dismissal, and event propagation scenarios.

Bug Fixes:
- Fix keyboard focus containment and restoration when drawers have disabled or missing footer controls.
- Correct Escape and backdrop dismissal behavior across modal and non-modal drawers.
- Improve scroll locking and restoration for modal drawers, including preservation of existing body styles.
- Prevent stale close animations and focus restoration from affecting drawers reopened during transitions.

Enhancements:
- Expose show and hide methods and document native dialog behavior, focus handling, and backdrop customization.
- Replace the legacy custom focus trap with the shared FocusTrap utility and expand drawer behavior coverage across desktop and mobile viewports.
- Deprecate the drawer backdrop CSS part in favor of token-based native ::backdrop customization.

Build:
- Update component, design-token, CLI, library, and stylesheet development dependencies.

Documentation:
- Update README code examples and API documentation for native dialog methods, accessibility behavior, and backdrop customization.
- Add a post-mortem documenting the native dialog migration and accessibility decisions.

Tests:
- Expand automated coverage for rendering, lifecycle, focus management, keyboard navigation, modal behavior, scroll locking, ARIA attributes, and native dialog integration.

Chores:
- Refresh generated documentation formatting and example initialization.

### CODE REVIEW

File: src/auro-floater-bib.js [hunk 1]
```diff
@@ -30,15 +31,46 @@ export class AuroFloaterBib extends LitElement {
...
+        );
+      }
+    });
   }
 
   render() {
     return html`
-      <div class="container">
+      <dialog class="container" aria-labelledby="dialogLabel">
+        <span id="dialogLabel" class="util_displayHiddenVisually" aria-hidden="true">${this.bibLabel || ""}</span>
```
[the remark this answers is not in the document]
Comment (sun-mota) [reply src/auro-floater-bib.js]: intentional change

File: test/auro-drawer.test.js [hunk 2]
```diff
@@ -1,249 +1,892 @@
...
+        const el = await fixture(html`
+          <auro-drawer open modal>
+            <h2 slot="header">It's a drawer</h2>
+            <div slot="content"><p>Hello World!</p></div>
+          </auro-drawer>
+        `);
+        window.dispatchEvent(new KeyboardEvent("keydown", { key: "Escape" }));
+        await elementUpdated(el);
+        expect(el.hasAttribute("open")).to.be.true;
+      });
```
[the remark this answers is not in the document]
Comment (sun-mota) [reply test/auro-drawer.test.js:307]: intended

File: test/auro-drawer.test.js [hunk 3]
```diff
@@ -1,249 +1,892 @@
...
+        const el = await fixture(html`
+          <auro-drawer open>
+            <h2 slot="header">It's a drawer</h2>
+            <div slot="content"><p>Hello World!</p></div>
+          </auro-drawer>
+        `);
+        document.dispatchEvent(new KeyboardEvent("keydown", { key: "Escape" }));
+        await elementUpdated(el);
+        expect(el.hasAttribute("open")).to.be.false;
+      });
```
[the remark this answers is not in the document]
Comment (sun-mota) [reply test/auro-drawer.test.js:291]: intended

File: package.json [hunk 4]
```diff
@@ -24,14 +24,15 @@
...
+    "@aurodesignsystem/auro-cli": "^3.7.1",
     "@aurodesignsystem/auro-config": "^1.3.1",
-    "@aurodesignsystem/auro-icon": "^9.1.1",
-    "@aurodesignsystem/auro-library": "^5.5.4",
-    "@aurodesignsystem/design-tokens": "^8.7.0",
-    "@aurodesignsystem/webcorestylesheets": "^10.1.4",
+    "@aurodesignsystem/auro-icon": "^9.2.0",
+    "@aurodesignsystem/auro-library": "^5.13.1",
+    "@aurodesignsystem/design-tokens": "^9.0.0",
+    "@aurodesignsystem/webcorestylesheets": "^11.1.3",
```
[the remark this answers is not in the document]
Comment (sun-mota) [reply package.json:33]: Our >=20 still covers that range. You shouldn't have to change anything.
 
Design token still has the old node requirements. It should ideally match our >=20. or at least be >=22

File: src/auro-floater-bib.js [hunk 5]
```diff
@@ -30,15 +31,65 @@ export class AuroFloaterBib extends LitElement {
...
+      }
+    });
   }
 
   render() {
     return html`
-      <div class="container">
+      <dialog class="container" aria-labelledby="dialogLabel">
+        <span id="dialogLabel" class="util_displayHiddenVisually">${this.bibLabel || ""}</span>
         <div class="backdrop" part="backdrop"></div>
```
[the remark this answers is not in the document]
Comment (sun-mota) [reply src/auro-floater-bib.js:60]: that's fine

File: src/styles/floaterBibStyles.scss [hunk 6]
```diff
@@ -14,30 +14,80 @@
...
+  }
+
+  // Decorative div kept for backwards-compatible ::part(drawer-backdrop) styling.
+  // pointer-events: none ensures clicks pass through to the dialog element
+  // so backdrop click-to-close detection (e.target === dialog) still works.
+  .backdrop {
+    position: fixed;
+    inset: 0;
+    pointer-events: none;
+  }
```
[the remark this answers is not in the document]
Comment (sun-mota) [reply src/styles/floaterBibStyles.scss:64]: looked into this and it turns out AuroFloatingUI sets contain: layout on the auro-drawer host element at runtime 
So in practice, .backdrop { position: fixed; inset: 0 } is already scoped  to auro-drawer's bounds rather than the viewport. Verified this in  DevTools where the backdrop renders at the bib's dimensions not full-screen. position: absolute would behave identically here, so we're keeping fixed as-is.

File: src/styles/drawerContentStyles.scss [hunk 7]
```diff
@@ -87,46 +86,61 @@
...
+// }
 
-@starting-style {
-  :host([visible][placement="right"]) .wrapper {
-    transform: translateX(100%);
+@mixin drawerPlacement($direction, $initial-offset, $transform-axis, $transform-offset) {
+  :host([placement="#{$direction}"]) .wrapper {
+    #{$direction}: #{$transform-offset};
+      transform: #{$transform-axis}(#{$initial-offset});
   }
```
[the remark this answers is not in the document]
Comment (sun-mota) [reply src/styles/drawerContentStyles.scss:132]: // Default (closed) state: panel is hidden off-screen.
      //
      // The direction property uses $transform-offset (e.g. right: 100%) to push
      // the panel into negative coordinate space — entirely past the opposing edge
      // of the container. The viewport always clips content at negative coordinates
      // (left of x=0, above y=0) for free, so no overflow:hidden is needed.
      //
      // The transform uses $initial-offset (0%) so it contributes no additional
      // offset; all hiding is done by the direction property alone.
      //
      // This is intentionally swapped from what you might expect ($direction: $initial-offset
      // + transform: $transform-offset). That original arrangement pushed right/bottom panels
      // into positive overflow (right of / below the container), which is NOT clipped by the
      // viewport and caused the panel to remain visible before the open animation started.

File: src/auro-drawer.js [hunk 8]
```diff
@@ -252,15 +253,22 @@ export class AuroDrawer extends AuroFloater {
...
+          AuroDrawer.properties[entry].attribute || entry,
+          this[entry],
+        );
       }
     });
 
     if (changedProperties.has("isPopoverVisible")) {
       this.drawerBib.visible = this.isPopoverVisible;
+      this.drawerBib.closing = !this.isPopoverVisible;
     }
```
[the remark this answers is not in the document]
Comment (sun-mota) [reply src/auro-drawer.js:273]: it's intended that his.drawerBib.closing being set based on the the changed value

File: src/auro-floater-bib.js [hunk 9]
```diff
@@ -30,15 +31,65 @@ export class AuroFloaterBib extends LitElement {
...
+    });
+
+    // Re-dispatch keydown events that stopped at the dialog boundary so that
+    // slotted consumer keyboard handlers outside the shadow DOM still receive them.
+    this.dialog.addEventListener("keydown", (e) => {
+      // Skip already-composed events (real user keystrokes cross shadow boundaries
+      // natively) and events whose target is not the dialog itself.
+      if (e.target !== this.dialog || e.composed) {
+        return;
+      }
```
[the remark this answers is not in the document]
Comment (sun-mota) [reply src/auro-floater-bib.js]: intended

**Review by jordanjones243 — CHANGES_REQUESTED** [review]
Noticed one bug still present. On mobile, when there is an input near the bottom of the drawer and you click on it, they keyboard opens and the drawer is shifted upwards and out of view. Looks like this was a bug listed as a part of this work. Everything else looks/works great.

File: src/auro-floater.js [hunk 10]
```diff
@@ -80,28 +79,130 @@ export class AuroFloater extends LitElement {
...
       } else {
-        this.floater.hideBib();
+        // Fallback for browsers without the Popover API
+        this.bib.dialog.removeAttribute("popover");
+        this.bib.dialog.show();
       }
+    } else {
+      this.bib.dialog.removeAttribute("popover");
+      this.bib.dialog.showModal();
+    }
```
Comment (jason-capsule42) [inline src/auro-floater.js:184]: ✅ **Resolved** — this finding no longer applies as of the current head; superseded by a newer review.

File: src/auro-floater.js [hunk 11]
```diff
@@ -80,28 +79,130 @@ export class AuroFloater extends LitElement {
...
+      }
+    }
+  }
+
+  /**
+   * Opens the native dialog inside the bib.
+   *
+   * - `modal && !nested`: `showModal()` for native focus containment and top-layer rendering.
+   * - `nested` or `!modal`: `showPopover()` to keep positional CSS intact
+   *   and allow free keyboard flow to background content (WCAG 2.1.2).
```
Comment (jason-capsule42) [inline src/auro-floater.js]: ✅ **Resolved** — this finding no longer applies as of the current head; superseded by a newer review.

**Comment by jason-capsule42** [comment]
## 🤖 Claude Code Review — native `<dialog>` migration (AB#1491978 / #133)

📊 **Review Quality:** Medium-high confidence. Diff is large (~3,800 changed lines, but ~750 of the churn is `package-lock.json`; source/test/docs is the meaningful portion), spanning 24 files. Cross-file analysis focused on `src/auro-floater*.js`, `auro-drawer*.js`, styles, and the test suite. Full context available: post-mortem `docs/post-mortem/1491978.md` read; **ℹ️ No TRD linked** (informational only). ⚠️ `node_modules` is not installed, so `@aurodesignsystem/auro-library` internals (`floatingUI.mjs`) could not be inspected — the Escape-handling finding below is therefore **PLAUSIBLE, not confirmed**. Estimated cost: **≈ 210k input tokens** across the two models (opus ≈ 133k, sonnet ≈ 78k) + orchestration.

**🤖 Multi-model review** — roster: **Claude Opus 4.8 + Claude Sonnet 5** (effort: high). This review always runs multi-model; findings below are reconciled with a corroboration gate (🔴 = either model; 🟡/📄 = both models).

---

### 🔴 Bug — `show()` rapid-reopen throws `InvalidStateError` *(opus, sonnet)*
Posted inline on `src/auro-floater.js`. `hide()` defers the real close 300ms; `show()` clears that timer, leaving the dialog/popover open, then calls `showModal()`/`showPopover()` on an already-open element → `InvalidStateError` as an unhandled rejection (`show()` is async and un-awaited from `updated()`). See the inline suggestion.

### 🔴 Bug (PLAUSIBLE — needs verification against the library) — no Escape-close path for non-modal / nested drawers *(opus only)*
The only Escape wiring is the `dialog-cancel` handler (`auro-floater.js:85`), and the native `<dialog>` `cancel` event fires **only for `showModal()`** (modal). Non-modal drawers open via `showPopover("manual")` — manual popovers do not respond to Escape — and nested drawers via `setAttribute("open")` — no `cancel`. So the `!modal → hide()` branch of the `dialog-cancel` handler appears unreachable, and default/nested drawers may have no Escape dismissal (WCAG 2.1.2). **However**, the WTR test `should close when ESC is pressed` (`test/auro-drawer.test.js:278`, real browser, default drawer) asserts exactly this behavior — if that test is green in CI, the Escape close is being provided by `AuroFloatingUI` (not inspectable here). **Action:** confirm against the installed library; if the library provides it, no code change is needed but the post-mortem §5/§10 wording (below) should still be corrected.

### 📄 Documentation — post-mortem describes a superseded design and a wrong default *(opus, sonnet)*
`docs/post-mortem/1491978.md`:
- **§2 / §10** say non-modal (non-nested) drawers use `dialog.setAttribute('open','')` and "a separate overlay element for visual backdrop" — the final code uses `showPopover()`.
- **§10** states "a manual `keydown` listener was added only for the non-modal path" — **no such listener exists** in the final code (Escape flows through the `cancel` event bridge).
- **§7** lists `--auro-drawer-backdrop-background` default as `rgba(0,0,0,0.5)`, but the actual CSS default is `transparent` (`src/styles/floaterBibStyles.scss:52`; `docs/partials/api.md:206` correctly says `transparent`).

Reconcile so the post-mortem doesn't mislead future maintainers. (No self-referencing pinned SHA present — that rule is not triggered.)

### 📄 Documentation — `show()` JSDoc & `docs/api.md` misdescribe the `nested` path *(opus, sonnet)*
Posted inline on the JSDoc. `docs/api.md` (the `show` method row) carries the same stale "`nested` or `!modal`: `showPopover()`" text and should be regenerated.

### 🔴 Commit Syntax — `test:` commit bundles a functional source fix *(opus only)*
Commit `test: expand unit test coverage with real-life use cases` (5ae8861) modifies `src/auro-drawer.js` / `src/auro-drawer-content.js` with a behavior fix ("Fix ariaLabel.drawer.close propagation … `requestUpdate` after slot content is moved"). Per Conventional Commits priority (`feat` > `fix` > others), a commit that mixes a bug fix with tests must use **`fix:`**, not `test:` — the wrong prefix suppresses the PATCH semver bump and misfiles the change in release notes.

### 🟡 Nit — commits missing a work-item / PR reference *(opus, sonnet)*
`e69b3f8` (fix: auro-library dev pre-release), `0d5ea4a` (perf: scroll lock), `4ae5185` (feat: consolidate show/hide), `bdccea9` (test: auro-input script source), and `5ae8861` (test: expand coverage) carry no `AB#` or `#133`/`#131` reference. Most other commits correctly cite `#133` / AB#1491978 — add references to these for ADO traceability.

---

### ✅ Checked and OK
- **No missing `BREAKING CHANGE`:** `show()`/`hide()` are additive; `::part(drawer-backdrop)` and the `.backdrop` div are retained for back-compat; removed `role="dialog"`/`aria-modal` are now provided natively by `showModal()`. No public-API break requiring a MAJOR bump.
- **Post-mortem present** at `docs/post-mortem/1491978.md`, correctly named to the ADO ticket — release-blocker requirement satisfied.
- `this.bib.bibLabel = this.triggerElement?.textContent.trim()` is null-safe (optional chaining short-circuits).
- Focus-restore / `closing`-reset timers are guarded; `disconnectedCallback` guards on `this.floater`; keydown re-dispatch bridge dispatches on the host (avoids prior recursion) and re-enumerates non-enumerable `KeyboardEvent` getters.

---

### 🤖 Model contribution summary
| Model | Raised | Survived | Unique | Corroborated |
|---|---|---|---|---|
| **Opus 4.8** | 6 | 6 | 2 (1 🔴 PLAUSIBLE Escape path, 1 🔴 Commit Syntax) | 3 (the 🔴 reopen bug + 2 📄 docs) |
| **Sonnet 5** | 5 | 4 | 1 🟡 (dead null-guard `auro-drawer-content.js:99` — un-corroborated, dropped from consensus per the both-models gate) | 3 (the 🔴 reopen bug + 2 📄 docs) |

**Caught by one model only:**
- **Opus** uniquely raised the 🔴 (PLAUSIBLE) missing Escape-close path for non-modal/nested drawers, and the 🔴 Commit Syntax prefix mismatch — sonnet did not surface either. These are opus's high-value contributions this run.
- **Sonnet** uniquely raised a 🟡 dead `else` null-guard in `ariaLabelDrawerClose` (`auro-drawer-content.js:99`); un-corroborated, so held out of consensus, noted here for the author's optional consideration.

**Verdict:** opus — 2 unique 🔴 (a plausible a11y regression + a semver-affecting commit-prefix error) → **high value this run**. sonnet — 0 unique 🔴, corroborated the 🔴 reopen bug and both 📄 doc findings (which is what promoted the docs findings past the both-models gate) → **solid corroboration value**.

*Both models independently identified the `show()` rapid-reopen `InvalidStateError` as the top finding.*

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]

**Review by jordanjones243 — DISMISSED** [review]

File: docs/api.md [hunk 12]
```diff
@@ -17,9 +17,11 @@ The `auro-drawer` element provides users a way to implement an expandable drawer
...
+| -------- | -------------------------------------------------------------------- | ------ | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| hide     | `eventType` (any)                                                    |        | Closes the native dialog.                                                                                                                                                                                                                                                                                                                                               |
```
Comment (sun-mota) [inline docs/api.md:22]: ✅ **Resolved** — this finding no longer applies as of the current head; superseded by a newer review.

**Comment by sun-mota** [comment]
## Claude Code Review — PR #131

### 📊 Review Quality

- **Diff size:** ~4,820 lines across 24 files — large diff; review was conducted in full.
- **Files touched:** 24 files.
- **Context availability:** Post-mortem `docs/post-mortem/1491978.md` found and read. TRD linked (`https://github.com/orgs/AlaskaAirlines/discussions/624`) but **could not be fetched** — discussion #624 does not exist at the `auro-drawer` repo level (likely an org-level discussion). TRD-deviation check was skipped.
- **Confidence:** Medium — large diff, post-mortem available, TRD unavailable.
- **Estimated token cost:** ≈ 55k input tokens (diff ~200k chars ÷ 4 + post-mortem ~8k chars + source files read ~10k chars).

---

### ℹ️ TRD Status

ℹ️ TRD linked (`https://github.com/orgs/AlaskaAirlines/discussions/624`) but could not be fetched — TRD-deviation check was skipped.

---

### Previous Findings — Now Resolved ✅

The two findings from the prior review (`e69b3f8`) are addressed in this head:
- ✅ **show() double-open `InvalidStateError`** — fixed in commit `6555d17` (added open-state guards before `showModal()`, `showPopover()`, and `dialog.show()`).
- ✅ **show() JSDoc / docs/api.md mismatch for nested path** — fixed in commit `6555d17` (corrected to three-path open strategy: nested / !modal / modal).

---

### New Findings

#### Inline (see thread)

- 📄 **Documentation (non-blocking):** `docs/api.md` — `hide()` method exposes `eventType (any)` as a documented public parameter. This is an internal parameter used only by `disconnectedCallback()` to signal lifecycle teardown; consumers should call `hide()` with no arguments. Full inline suggestion posted on line 22.

#### Summary-level

**Commit message traceability — 🟡 Nit:**
Four early commits are missing an `AB#` ticket reference and do not reference PR #131. They have `Resolves: #133` (a linked issue) but not a work-item reference. Per project convention, all commits should be traceable to an ADO ticket:
- `70cbf35` — `feat(backdrop): migrate to native dialog::backdrop with passthrough tokens`
- `184c8b5` — `fix(scroll): use position:fixed body lock to prevent VoiceOver scroll-through`
- `e6ec169` — `fix: migrate drawer bib to native <dialog> element`
- `1cb7230` — `test: add focus management tests for auro-drawer`

These represent the initial work for this migration; subsequent commits all carry `AB#1491978`. The missing references don't block merge but are worth noting for traceability.

---

### Post-Mortem Validation ✅

- `docs/post-mortem/1491978.md` exists, filename matches `AB#1491978`. ✅
- No pinned SHAs in post-mortem prose. ✅
- TRD deviation check: skipped (TRD unavailable). ℹ️

---

### 🤖 Multi-Model Review

This review was conducted by this orchestrating agent as a comprehensive multi-persona analysis (Consumer developer, Framework integrator, Accessibility auditor, Performance engineer, Security reviewer, QA engineer, Future maintainer, Release manager, Staff engineer) applied in a single pass against the full diff, post-mortem, and all source files.

**Model roster:** Claude Sonnet 4.6 (orchestrator, performing direct review) — Note: subagent fan-out was not available in this session (sandbox restrictions blocked the Task tool pattern); the orchestrator performed the full review directly.

**Key findings investigated and cleared:**
- Animation reset race between `show()` (async, awaits `bib.updateComplete`) and CSS keyframes replay — no issue; `show()` awaits before opening the dialog, so the animation reset in `updated()` runs first.
- `[visible]` + `[closing]` CSS specificity conflict (same selector weight, `[closing]` wins) — correctly handled: `closing = false` is set synchronously before `visible = true` in the same Lit render batch.
- `_closeTimeout` / `_showGeneration` race on rapid open/close — correctly handled by `clearTimeout` + generation counter pattern.
- Non-modal popover hide path (`dialog.open` is never set by `showPopover()`) — correctly guarded with `:popover-open` check before `hidePopover()`.
- FocusTrap lifecycle on disconnect — handled via `visible = false` → `updated()` → `focusTrap.disconnect()` chain (async but safe; element is already disconnected by the time it matters).
- Memory leak from anonymous event listeners in `firstUpdated()` — not a leak; `this.bib` is shadow-DOM-scoped and GC'd with the host.
- `FocusTrap` deletion from `src/util/FocusTrap.js` — not a breaking change; `FocusTrap` was not exported from `src/index.js`.
- Backdrop breaking change for modal drawers — no change; the `.backdrop` div scrim behavior is preserved for existing consumers.
- `feat(backdrop)` and `feat: consolidate` commits without `BREAKING CHANGE` — correct; these are additive changes with backwards-compatible deprecation path.

**Unique contribution:**
The only new finding is the `hide(eventType)` API docs issue (📄 non-blocking). No 🔴 bugs remain open.

---

### Overall Verdict

✅ **No blocking issues.** The single open finding is a non-blocking docs accuracy item (`hide()` parameter). The core migration is well-implemented: proper three-path open strategy, correct CSS animation lifecycle, robust race-condition handling, and thorough test coverage for the key edge cases (rapid open/close, focus restore timing, non-modal popover teardown).

File: docs/api.md (hunk 12 above)
Comment (sun-mota) [reply docs/api.md:22]: won't fix

File: test/auro-drawer.test.js [hunk 13]
```diff
@@ -1,249 +1,1022 @@
...
+          <div style="width: 300px; height: 300px;">
+            <auro-drawer open nested>
+              <h2 slot="header">Nested</h2>
+              <div slot="content"><p>Content</p></div>
+            </auro-drawer>
+          </div>
+        `);
+        const drawer = el.querySelector("auro-drawer");
+        await elementUpdated(drawer);
+        expect(drawer.bib._scrollLocked, "nested drawer must not lock scroll").to.not.be.true;
```
Comment (jason-capsule42) [inline test/auro-drawer.test.js]: 🔴 **Bug:** This assertion reads the wrong object, so it provides **zero coverage**. `_scrollLocked` is a property of `AuroFloatingUI` — which is `drawer.floater`, not `drawer.bib` (`AuroFloaterBib`). Every sibling test uses `el.floater._scrollLocked` (see the `modal` and scroll-lock-release tests). Because `drawer.bib._scrollLocked` is always `undefined`, `.to.not.be.true` passes vacuously and the "nested drawer must not lock scroll" guarantee is never actually exercised.

```suggestion
        expect(drawer.floater._scrollLocked, "nested drawer must not lock scroll").to.not.be.true;
```
_(found by: sonnet)_

File: src/auro-floater.js [hunk 14]
```diff
@@ -80,28 +84,143 @@ export class AuroFloater extends LitElement {
...
+   *
+   * - `nested`: `setAttribute("open", "")` to anchor within the parent container.
+   * - `!modal && !nested`: `showPopover()` to keep positional CSS intact
+   *   and allow free keyboard flow to background content (WCAG 2.1.2).
+   * - `modal && !nested`: `showModal()` for native focus containment and top-layer rendering.
+   */
+  async show() {
+    const generation = ++this._showGeneration;
+    clearTimeout(this._closeTimeout);
+    this.floater.showBib();
```
Comment (jason-capsule42) [inline src/auro-floater.js:151]: 🔴 **Bug:** `show()` is a documented public method (see `docs/api.md`), but it calls `this.floater.showBib()` with no guard for an uninitialized `this.floater`. `this.floater` is only set in `firstUpdated()`; `disconnectedCallback()` already guards with `if (this.floater)`, but `show()` does not. Calling `el.show()` synchronously after `document.body.appendChild(el)` — before Lit's first update cycle — throws `TypeError: Cannot read properties of undefined (reading 'showBib')`. A bare `return` would silently no-op the open, so awaiting the first update is safer:

```suggestion
    if (!this.floater) {
      await this.updateComplete;
    }
    this.floater.showBib();
```
_(found by: sonnet)_

**Review by jason-capsule42 — DISMISSED** [review]

**Review by jason-capsule42 — DISMISSED** [review]

**Review by jason-capsule42 — APPROVED** [review]
